File: Java/217. Contains Duplicate.java
/**
 * https://leetcode.com/problems/contains-duplicate/submissions/
 */

import java.util.HashSet;
import java.util.Set;

class Solution {
    // O(n) time complexity
    // O(n) space complexity
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums)
            set.add(num);

        return set.size() != nums.length;
    }
}

File: Kotlin/217. Contains Duplicate.kt
/**
 * https://leetcode.com/problems/contains-duplicate/
 */

class Solution {
    // O(n) time complexity
    // O(n) space complexity
    fun containsDuplicate(nums: IntArray): Boolean {
        val set = mutableSetOf<Int>()
        nums.forEach {
            set.add(it)
        }
        return set.size != nums.size
    }
}